_episodes/08-func.md

```diff
@@ -177,6 +177,31 @@ temperature in Kelvin was: 373.15
 ~~~
 {: .output}
 
+On the contrary, inside a function, one can read the value of a variable which has been defined outside the function, before the function is called. We refer to these variables as [global variables]({{ page.root }}/reference.html#global-variable).
+
+In the modified example below, we check the value of the global variable `temp_kelvin` inside a function.
+
+~~~
+def check_temp_kelvin():
+    print('temperature in Kelvin was:', temp_kelvin)
+
+check_temp_kelvin()
+temp_kelvin = fahr_to_kelvin(32.0)
+check_temp_kelvin()
+~~~
+{: .language-python}
+
+~~~
+temperature in Kelvin was: 373.15
+temperature in Kelvin was: 273.15
+~~~
+{: .output}
```

Comment:
I think this would still make sense if it were shorter. What do you think about taking out the
```
temp_kelvin = fahr_to_kelvin(32.0)
check_temp_kelvin()
```
and 
```
temperature in Kelvin was: 273.15
```

Reply:
I was vaguelay fearing that a beginner may think that the value of `temp_kelvin` is read once for all when the interpreter is going throug the function definition, then reused as is, what new value the variable may get. 

Another misocneption may be that the global variable must exist before the function definition is parsed.

Do you think all those fears are excessive, given all the material that the learner got in previous episodes ?

Reply:
Regarding trying to debunk the misconception that the global variable must exist before the function definition is parsed, since there will still already have been a value existing with the name `temp_kelvin` before the function definition, I am not sure how much reassigning it after the definition and first call of `check_temp_kelvin` clarifies that for a beginner. I think to do that, the function could be defined to read a global variable that has not yet been named, then create the variable, then call the function:

```
def print_kelvin():
    print('temperature in Kelvin was:', global_kelvin)

global_kelvin = fahr_to_kelvin(32.0)
print_kelvin()
```

Reply:
Yes. And perhaps it would make more clear that the variable is searched for when executing a function call, then fighting also the first misconception. Ok. More to come.